be/src/util/byte_buffer.h

```diff
@@ -113,13 +182,26 @@ struct ByteBuffer {
     size_t remaining() const { return limit - pos; }
     bool has_remaining() const { return limit > pos; }
 
+    ByteBufferMeta* meta() { return _meta; }
+
     char* const ptr;
     size_t pos{0};
     size_t limit;
     size_t capacity;
 
 private:
-    ByteBuffer(size_t capacity_) : ptr(new char[capacity_]), limit(capacity_), capacity(capacity_) {}
+    ByteBuffer(size_t capacity_, ByteBufferMeta* meta = NoneByteBufferMeta::instance())
+            : ptr(new char[capacity_]), limit(capacity_), capacity(capacity_), _meta(meta) {
+        DCHECK(_meta != nullptr);
+    };
+
+    static void delete_meta_safely(ByteBufferMeta* meta) {
+        if (meta != nullptr && meta != NoneByteBufferMeta::instance()) {
+            delete meta;
+        }
+    }
+
+    ByteBufferMeta* _meta;
```

Comment:
can this `_meta` managed by shared_ptr or unique_ptr? shall avoid managing the raw pointer whenever possible.

Reply:
do not use shared_ptr/unique_ptr because not all `ByteBuffer` need meta and raw pointer is more lightweight for those buffers. Here raw pointer for metas is managed in the same style as the data raw pointer